container/go/pkg/compat/index.go

```diff
@@ -0,0 +1,198 @@
+// Naively validates a legacy intermediate layout at <path> by checking if digest, config.json, and manifest.json all exist.
+func FromPath(path string) (Path, error) {
+	var err error
+	_, err = os.Stat(filepath.Join(path, manifestFile))
+	if err != nil {
+		return "", err
+	}
+
+	_, err = os.Stat(filepath.Join(path, configFile))
+	if err != nil {
+		return "", err
+	}
+
+	_, err = os.Stat(filepath.Join(path, digestFile))
+	if err != nil {
+		return "", err
+	}
+
+	return Path(path), nil
+}
+
+// This intermediate layout implements v1.ImageIndex.
+type intermediateLayout struct {
+	// path of this layout, with helper functions for finding the full directory.
+	path Path
+	// manifestLock protects raw manifest.
+	manifestLock sync.Mutex
+	// rawManifest is the raw bytes of manifest.json file.
+	rawManifest []byte
+}
+
+// MediaType of this image index's manifest.
+func (i *intermediateLayout) MediaType() (types.MediaType, error) {
+	// TODO: This image index does not follow the OCI standards, but the contents are compatible indeed.
+	// We will have this image index as an OCIImageIndex type for now, since this is only a intermediate format.
+	return types.OCIImageIndex, nil
+}
+
+// Digest returns the sha256 hash of this index's manifest.json metadata, an entrypoint for the config and layers.
+func (i *intermediateLayout) Digest() (v1.Hash, error) {
+	// We expect a file named digest that stores the manifest's hash formatted as sha256:{Hash} in this directory.
+	digest, err := ioutil.ReadFile(i.path.path(digestFile))
+	if err != nil {
+		fmt.Errorf("Failed to locate SHA256 digest file for image manifest: %v", err)
+	}
+
+	return v1.NewHash(string(digest))
+}
+
+// IndexManifest returns this image index's manifest object, simulates index.json in OCI layout.
+func (i *intermediateLayout) IndexManifest() (*v1.IndexManifest, error) {
+	// Parse raw manifest into a manifest struct.
+	manifest, err := partial.Manifest(i)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse raw manifest, please check if a correctly formatted manifest.json exists %v", err)
+	}
+	manifestDigest, err := i.Digest()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse image manifest hash, please check if a digest file exists in the directory and it is formatted as {Algorithm}:{Hash} %v", err)
+	}
+
+	// We are missing index.json in this intermediate format.
+	// Since index.json is represented in IndexManifest structure, we will populate this struct with parsed info.
+	index := v1.IndexManifest{
+		SchemaVersion: manifest.SchemaVersion,
+		Manifests: []v1.Descriptor{
+			v1.Descriptor{
+				MediaType: manifest.MediaType,
+				Size:      int64(len(i.rawManifest)),
+				Digest:    manifestDigest,
+			},
+		},
+	}
+
+	return &index, nil
+}
+
+// RawManifest returns the serialized bytes of manifest.json metadata.
+func (i *intermediateLayout) RawManifest() ([]byte, error) {
```

Comment:
This seems the manifest of the image. But layout's RawManifest is the one for index.json.

Reply:
Return an error when it's called.